yelp/client.py

```diff
@@ -11,35 +10,13 @@
 from yelp.errors import ErrorHandler
 
 
-class Client(object):
-
-    _endpoints = [
-        Business,
-        PhoneSearch,
-        Search
-    ]
+class BaseClient(object):
 
     def __init__(self, authenticator):
         self.authenticator = authenticator
         self._error_handler = ErrorHandler()
-        self._define_request_methods()
-
-    # Creates an instance of each endpoint class and adds the instances' public
-    # singleton methods to Client. We do this to promote modularity.
-    def _define_request_methods(self):
-        endpoint_instances = [end(self) for end in self._endpoints]
-        for endpoint in endpoint_instances:
-            instance_methods = inspect.getmembers(endpoint, inspect.ismethod)
-            self._add_instance_methods(instance_methods)
-
-    def _add_instance_methods(self, instance_methods):
-        # instance_methods is a list of (name, value) tuples where value is the
-        # instance of the bound method
-        for method in instance_methods:
-            if method[0][0] is not '_':
-                self.__setattr__(method[0], method[1])
-
-    def _make_request(self, path, url_params={}):
+
+    def make_request(self, path, url_params={}):
```

Comment:
Perhaps this should be left as `_make_request` to discourage people from using it directly.